engine/src/flutter/shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -18,12 +18,58 @@
  */
 public interface TextureRegistry {
   /**
-   * Creates and registers a SurfaceProducer texture managed by the Flutter engine.
+   * Creates and registers a {@link SurfaceProducer}, or a Flutter-managed {@link Surface}.
+   *
+   * <p>Uses the {@link SurfaceLifecycle#manual} lifecycle implicitly.
    *
    * @return A SurfaceProducer.
    */
   @NonNull
-  SurfaceProducer createSurfaceProducer();
+  default SurfaceProducer createSurfaceProducer() {
+    return createSurfaceProducer(SurfaceLifecycle.manual);
+  }
+
+  /**
+   * Creates and a {@link SurfaceProducer}, or a Flutter-managed {@link Surface}.
+   *
+   * @param lifecycle Whether to automatically reset the last image and release the surface.
+   * @return A SurfaceProducer.
+   */
+  @NonNull
+  SurfaceProducer createSurfaceProducer(SurfaceLifecycle lifecycle);
+
+  /**
+   * How a {@link SurfaceProducer} created by {@link #createSurfaceProducer()} manages the lifecycle
+   * of the created surface.
+   */
+  enum SurfaceLifecycle {
+    /**
+     * The surface and latest image should be kept, even if the app enters the background.
+     *
+     * <p>The application, or calling code, can choose to (manually) reset the surface at the
+     * appropriate time (such as to lower memory pressure, or cleanup an unused surface), but by
+     * default the surface will never be reset, and as a result, new images do not have to be drawn
+     * to the surface.
+     *
+     * <p>This is an appropriate lifecycle for external textures, as it is not guaranteed that new
+     * images will be drawn to the surface, and whether the image should be kept when the app is
+     * backgrounded.
+     */
+    manual,
+
+    /**
+     * The surface may be reset if the app enters the background.
```

Comment:
This has to be `will` and not `may`, because if the image is cleared and the surface is not reset, we'll get the bug you're trying to fix.